src/elements/Segment/Segments.js

```diff
@@ -0,0 +1,46 @@
+import React, {Children, Component, PropTypes} from 'react';
+import _ from 'lodash';
+import invariant from 'fbjs/lib/invariant';
+import classNames from 'classnames';
+import META from 'src/utils/Meta';
+import Segment from './Segment';
+
+export default class Segments extends Component {
+  static propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+  };
+
+  componentDidMount() {
+    const {children} = this.props;
+
+    invariant(
+      !_.any( Children.map(children, child => child.type !== Segment) ),
+      'May only contain children of type Segment.',
+    );
```

Comment:
@levithomason this would be our first use of `fbjs` to throw a possible `invariant` violation in this repo. I also tried to create an `invariant` for `Children(count) >= 2` but unsuccessful. The intention here is to help prevent misuse of the components we're creating without first creating unexpected side effects such as node tree munging.

If we like the pattern we could also consider adding a runtime check for children, though that's something we may want to pair on to get out the door faster.